### PR TITLE
feat: detect xcframework type to determine how to import

### LINKED ISSUE #559: Allow a client to specify how a `binaryTarget` is linked
When support for `binaryTarget` targets was added to this repository, they were modeled as `apple_dynamic_xcframework_import` Bazel targets. However, with #498, the implementation was changed to model `binaryTarget` as `apple_static_xcframework_import`.  Ideally, clients could select how these targets should be exposed.

The following is the code that was used to expose the binary targets as a dynamic xcframework.

```python
# Code from swikftpkg_build_files.bzl

def _apple_dynamic_xcframework_import_build_file(target):
    load_stmts = [apple_dynamic_xcframework_import_load_stmt]
    glob = scg.new_fn_call(
        "glob",
        ["{tpath}/*.xcframework/**".format(tpath = target.path)],
    )
    decls = [
        build_decls.new(
            kind = apple_kinds.dynamic_xcframework_import,
            name = pkginfo_targets.bazel_label_name(target),
            attrs = {
                "visibility": ["//visibility:public"],
                "xcframework_imports": glob,
            },
        ),
    ]
    return build_files.new(
        load_stmts = load_stmts,
        decls = decls,
    )
```

Related to #498.
Related [Slack thread](https://bazelbuild.slack.com/archives/C05NZSHE7FU/p1693060367833069).

### PR DESCRIPTION
**Pull request by cgrindel**
- Add code to detect how a downloaded xcframework should be imported.
- Add `lottie_ios_example`. This example downloads a dynamic xcframework from [lottie-ios](https://github.com/airbnb/lottie-ios) via [lottie-spm](https://github.com/airbnb/lottie-spm).

Closes #559.

### CODE REVIEW

**Comment by cgrindel** [comment]
cc: @jpsim

**Review by jpsim — APPROVED** [review]
+1 seems to fix my project

File: examples/lottie_ios_example/swift_deps_index.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      "package_identity": "lottie-spm",
+      "product_memberships": [
+        "Lottie"
+      ]
+    },
+    {
+      "name": "Lottie",
+      "c99name": "Lottie",
+      "src_type": "binary",
+      "label": "@swiftpkg_lottie_spm//:remote_archive_Lottie.xcframework.zip_Lottie",
```
[the remark this answers is not in the document]
Comment (cgrindel) [reply examples/lottie_ios_example/swift_deps_index.json:20]: I created #585 to track the shortening of label.

File: swiftpkg/internal/artifact_infos.bzl [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    Args:
+        path: The path to the `XXX.framework` directory as a `string`.
+
+    Returns:
+        The framework name as a `string`.
+    """
+    basename = paths.basename(path)
+    (name, ext) = paths.split_extension(basename)
+    if ext != ".framework":
+        fail("The path does not point to an Apple framework. path: {}".format(path))
```
[the remark this answers is not in the document]
Comment (cgrindel) [reply swiftpkg/internal/artifact_infos.bzl:58]: Good question. The code expects an xcframework which contains one or more frameworks. There really isn't anything that the client can do to fix this. If we get to this failure, it means that something is wrong with the xcframework OR `rules_swift_package_manager` does not support whatever was just downloaded. 

I am open to other language. We could ask them to file a bug at `rules_swift_package_manager`. What do you think?

[the remark this answers is not in the document]
Comment (luispadron) [reply swiftpkg/internal/artifact_infos.bzl]: Yep that seems good to me

[the remark this answers is not in the document]
Comment (cgrindel) [reply swiftpkg/internal/artifact_infos.bzl]: I went ahead and updated the message asking the client to file a bug. There may be other spots in the code where we want to do the same. We can consider propagating this pattern in future PRs.

**Comment by cgrindel** [comment]
I ended up removing the failure message if the binary target is not an xcframework. I was not thinking about #558 when I wrote that code.

**Pull request merged**